### DISCUSSION TITLE
Azure KeyVault reference w konfiguracji aplikacji.

### DISCUSSION THREAD

**Discussion by MichalKosowski** [AzureCommunityPL/group#49] [Community Content]
Hej,

Jakiś czas temu wrzuciłem info o moim debiucie blogowym na medium na grupę .Net Dev Poland. Jak słusznie zauważono - bardziej nadaje się to na grupę Azure. Także wrzucam info tutaj, może komuś się przyda (jak nie to napewno zmusi mnie to dalszego pisania 😅):
-> https://medium.com/@kosowski.michal/the-secure-way-to-pass-secrets-to-your-azure-app-5dfdd99a8e88

(mam też dwa arty o APIM przekopiowane z firmowego bloga, ciągle próbuję skończyć pisać o automatyzacji tego cholerstwa, ale priorytet spedł...)

I tak wiem - komletnie pominąłem dostęp przez AD, o którym nie wiedziałem. Ale też nie ma on zastosowania w scenariuszu który rozważałem (np. Function Triggers). Napewno będzie jakiś update jak znowu znajdę chwilkę aby coś skrobnąć :) Jedno jest pewne - dzięki temu wpisowi uporządkowałem swoją więdzę oraz dowiedziałem się nowych rzeczy dzięki komentarzą innych. Czyli to co najważniejsze :)

**Comment by TomaszOledzki** [comment]
Masz doświadczenie z referencjami do KV w FunctionApp?

Gdy dla "AzureWebJobsStorage" używam referencji do KV z Connection Stringiem do Storage, to czasem (nie zawsze) FunctionApp potrafi nie wstać. Jeśli dla tego konkretnego App Settinga wrzucam Connection String do Storage to FunctionApp nigdy nie ma problemów ze startem.

**Reply by MichalKosowski** [reply]
Nie zauważyłem u siebie takich problemów.... Ale connection string do Storage wrzucam w normalne parametry konfiguracyjne aplikacji a nie sekcje ConnectionStrings.

**Reply by TomaszOledzki** [reply]
AzureWebJobsStorage trzymam w App Settings, nie w sekcji Connection Strings.

**Reply by kamil-mrzyglod** [reply]
@TomaszOledzki - zgłaszałeś buga do PG? To generalnie powinno działać out-of-the-box, ale w przypadku Function App widziałem już tyle baboli, że wszystkiego bym się spodziewał.
